docs/rfcs/0002-session-id-sharing.md

```diff
@@ -0,0 +1,223 @@
+   any lingering data before starting collection telemetry related to the new session.
+2. The `shared_session_id` is generated and recorded.
+3. The value recorded to telemetry is also persisted on the daemon's local storage.
+4. Session metrics may be recorded from this point on.
+5. Session is started.
+
+The `flush` ping submitted on step one is expected to be empty. The submission of it right
+before the start of a new session is a safety measure to guarantee no lingering data makes it
+to the actual session ping.
+
+#### On session end
+
+Once an ongoing session is ended,
+
+1. The `daemonsession` ping is submitted with the reason `daemon_end`.
+2. The daemon's local storage for the `shared_session_id` is cleared.
+3. The `shared_session_id` metric is set to a known value. Due to it's `User` lifetime,
+   this metric cannot be cleared from storage ever. To indicate a period of inactivity,
+   we will leverage a known UUID.
+
+#### On daemon process start
+
+The session end trigger may not be hit, if the daemon process is force killed. Therefore,
+whenever the daemon process is started the local storage will be searched for a lingering
+`shared_session_id` value. That is a clear signal that the session end trigger was not hit
+and that there may be lingering data from a previous session on the Glean storage.
+
+When that is the case,
+
+1. The lingering `shared_session_id` value is recorded to Glean.
+2. The `daemonsession` ping is submitted with the reason `flush_init`.
+
+### On the main application
+
+The same three triggers which are important for the daemon and the `daemonsession` ping,
+are also important for the main application and the `vpnsession` ping.
+
+#### On session start
+
+If the main application is active during a session start it will,
+
+1. Signal to the daemon that a new session is to be started.
+2. It will then ask the daemon for the value of the shared session id it created.
+3. If the previous two communications where successful
+   1. The `vpnsession` ping will be submitted with reason `flush`.
+   2. The value of the retrieved `shared_session_id` is recorded.
+   3. Other session metrics may be recorded from this point.
+   4. If the communication with the daemon fails, an error metric is recorded.
+
+> **Note**: If the retrieved session id is the known inactive session id,
+> this is considered an unsuccessful retrieval and the error path is taken.
+
+#### On session end
+
+If the main application is active during a session end it will,
+
+1. Submit the session ping with reason `end`.
+2. Set `shared_session_id` metric is set to a known value. The same known value used for inactive
+   sessions on the `daemonsession` ping.
+
+#### On application start
+
+When the main application is started, just like in the daemon, there may be some lingering data
+from a previous session in storage. Om the main application, the session_id is not directly managed,
```

Comment:
```suggestion
from a previous session in storage. On the main application, the session_id is not directly managed,
```

Reply:
LOL, it's funny the spellchecker did not catch this.